fw/http_frame.c

```diff
@@ -2230,9 +2247,23 @@ tfw_h2_make_frames(struct sock *sk, TfwH2Ctx *ctx, unsigned long snd_wnd,
 		 * active stream.
 		 */
 		BUG_ON(!stream);
-		r = tfw_h2_stream_xmit_process(sk, ctx, stream, &snd_wnd);
-		deficit = tfw_h2_stream_recalc_deficit(stream);
-		tfw_h2_sched_stream_enqueue(sched, stream, parent, deficit);
+		r = tfw_h2_stream_xmit_process(sk, ctx, stream, !parent,
+					       &snd_wnd);
+
+		/* We don't recalculate deficits of exclusive streams. */
+		if (parent) {
+			deficit = tfw_h2_stream_recalc_deficit(stream);
+			tfw_h2_sched_stream_enqueue(sched, stream, parent,
+						    deficit);
+		} else if (!tfw_h2_stream_is_active(stream)) {
```

Comment:
Could you please place `else if ()` on separate line to make the conditions separate from the code above and more readable?